File: Day5/solution_part1.py
#!/usr/bin/env python

with open('input.txt') as f:
    content = [int(x) for x in f.readlines()]

counter = 0
index = 0
while index < len(content):
    offset = content[index]
    content[index] += 1
    index += offset
    counter += 1

print counter

File: Day6/solution_part1.rb
#!/usr/bin/env ruby
require 'set'

set = Set.new []
f = File.read("input.txt").split("\t").map(&:to_i)
set.add(f.join(','))

old_size = set.size
new_size = old_size
counter = 0
begin
    val = f.max
    indx = f.index(val)
    f[indx] = 0

    while val != 0
        indx += 1
        if indx >= f.length 
            indx = indx - f.length
        end
        f[indx] += 1
        val -= 1
    end

    set.add(f.join(','))
    old_size = new_size
    new_size = set.size
    counter+= 1

end while old_size != new_size

puts counter